diff --git a/libs/ui/Android.bp b/libs/ui/Android.bp
index 73adb0ae2..bab7b3270 100644
--- a/libs/ui/Android.bp
+++ b/libs/ui/Android.bp
@@ -27,6 +27,7 @@ cc_library_shared {
     ],
     cppflags: [
         "-Wextra",
+        "-Werror",
     ],
 
     sanitize: {
diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 878761b07..0bcdd1cd7 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -397,6 +397,9 @@ SurfaceFlinger::SurfaceFlinger(Factory& factory) : SurfaceFlinger(factory, SkipI
     property_get("debug.sf.showupdates", value, "0");
     mDebugRegion = atoi(value);
 
+    property_get("debug.sf.disable_hwc", value, "0");
+    mDebugDisableHWC = atoi(value);
+
     ALOGI_IF(mDebugRegion, "showupdates enabled");
 
     // DDMS debugging deprecated (b/120782499)
